ambry-api/src/main/java/com.github.ambry/router/Router.java

```diff
@@ -49,12 +49,15 @@
   public Future<ReadableStreamChannel> getBlob(String blobId);
 
   /**
-   * Requests for the blob data asynchronously and invokes the {@link Callback} when the request completes.
+   * Requests for the blob data asynchronously with user-set {@link GetBlobOptions} and invokes the {@link Callback}
+   * when the request completes.
    * @param blobId The ID of the blob for which blob data is requested.
+   * @param options The {@link GetBlobOptions} associated with the request.
    * @param callback The callback which will be invoked on the completion of the request.
    * @return A future that would contain a {@link ReadableStreamChannel} that represents the blob data eventually.
    */
-  public Future<ReadableStreamChannel> getBlob(String blobId, Callback<ReadableStreamChannel> callback);
+  public Future<ReadableStreamChannel> getBlob(String blobId, GetBlobOptions options,
```

Comment:
if you are changing this API, you need to change the `getBlob(String)` API also to `getBlob(String, GetBlobOptions)`